ex2/ex_2.cs

```diff
@@ -97,56 +92,63 @@ static void Main()
             {
                 owner = "1",
                 moneyInAccount = 10,
-                accType = AccountType.Checking
+                accountType = AccountType.Checking
             };
-            BankAccount acc2 = new BankAccount
+            BankAccount account2 = new BankAccount
             {
                 owner = "2",
                 moneyInAccount = 20,
-                accType = AccountType.Checking
+                accountType = AccountType.Checking
             };
-            Console.WriteLine(SumOfBalances(acc1, acc2));
+            Console.WriteLine(SumOfBalances(account1, account2));
+            BankAccount[] noAccounts = null;
+            Console.WriteLine(SumOfBalances(noAccounts));  // If null, returns 0.
             // 5
             int numAccounts = 10;
             List<BankAccount> accounts = new List<BankAccount>();
             for (int i=0; i<numAccounts; i++)
             {
-                BankAccount tempAccount = new BankAccount();
-                tempAccount.owner = $"Owner Number {i + 1}";
-                Random r1 = new Random(i);
-                tempAccount.moneyInAccount = r1.Next(1000);
-                tempAccount.accType = AccountType.Deposit;
+                BankAccount tempAccount = new BankAccount()
+                {
+                    owner = $"Owner Number {i + 1}",
+                    moneyInAccount = new Random(i).Next(1000),
+                    accountType = AccountType.Deposit
+                };
                 accounts.Add(tempAccount);
             }
             Console.WriteLine("Money in accounts before sorting:");
-            foreach (BankAccount bac in accounts) Console.Write($"{bac.moneyInAccount} ");
+            foreach (BankAccount bankAccount in accounts) Console.Write($"{bankAccount.moneyInAccount} ");
             Console.Write("\n");
-            GetRichestAccount(accounts);
-            GetPoorestAccount(accounts);
+            BankAccount richest = GetRichestAccount(accounts);
+            Console.WriteLine("The richest account:");
+            Console.WriteLine(richest.ToString());
+            BankAccount poorest = GetPoorestAccount(accounts);
+            Console.WriteLine("The poorest account:");
+            Console.WriteLine(poorest.ToString());
             accounts.Sort();
             Console.WriteLine("Sorted money in accounts:");
-            foreach (BankAccount bac in accounts) Console.Write($"{bac.moneyInAccount} ");
+            foreach (BankAccount bankAccount in accounts) Console.Write($"{bankAccount.moneyInAccount} ");
             Console.Write("\n");
             // Test 6
-            AddMoney(ref account, 10);
-            PrintAccount(account);
+            AddMoney(ref account0, 10);
+            Console.WriteLine(account0.ToString());
             // Test 7
-            account += 10;
-            PrintAccount(account);
+            account0 += 10;
+            Console.WriteLine(account0.ToString());
```

Comment:
You do not have to explicitly call `ToString` here, the function `Console.WriteLine` does in implicitly. Read the guide of casting https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/types/casting-and-type-conversions